analyzers/src/SonarAnalyzer.Common/Rules/DoNotCheckZeroSizeCollectionBase.cs

```diff
@@ -33,21 +34,28 @@ public abstract class DoNotCheckZeroSizeCollectionBase<TLanguageKindEnum,
         where TExpressionSyntax : SyntaxNode
     {
         protected const string DiagnosticId = "S3981";
-
-        protected const string MessageFormat =
+        private const string MessageFormat =
             "The {0} of '{1}' is always '>=0', so fix this test to get the real expected behavior.";
 
+        private readonly DiagnosticDescriptor rule;
+
         protected abstract TLanguageKindEnum GreaterThanOrEqualExpression { get; }
         protected abstract TLanguageKindEnum LessThanOrEqualExpression { get; }
         protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+        protected abstract string IEnumerableTString { get; }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected DoNotCheckZeroSizeCollectionBase(System.Resources.ResourceManager rspecStrings) =>
```

Comment:
pedantic: we can add an using for `System.Resources`

Reply:
This is a common pattern for the base classes.